mlx/traceability.py

```diff
@@ -207,6 +213,59 @@ def run(self):
         return [targetnode, itemnode]
 
 
+class ItemAttributeDirective(Directive):
+    """
+    Directive to declare attribute for items
+
+    Syntax::
+
+      .. item-attribute:: attribute_id [attribute_caption]
+
+         [attribute_content]
+
+    """
+    # Required argument: id
+    required_arguments = 1
+    # Optional argument: caption (whitespace allowed)
+    optional_arguments = 1
+    final_argument_whitespace = True
+    # Content allowed
+    has_content = True
+
+    def run(self):
+        env = self.state.document.settings.env
+
+        # Convert to lower-case as sphinx only allows lower case arguments (attribute to item directive)
+        attrid = self.arguments[0].lower()
```

Comment:
Can we put this check somewhere at plugin init/start and print a warning? This is the second time we are converting something for the user.